app/directives/formats/forms/national-report-6.js

```diff
@@ -445,10 +447,85 @@ define(['require', 'text!./national-report-6.html', 'app', 'angular', 'lodash',
 						//
 						//==================================
 						$scope.onPrePublish = function() {
-							return $scope.validate().then(function(hasError) {
+							$scope.subRecordErrors = undefined;
+							$scope.colorSubRecordError();
+							return $scope.validate()
+							.then(function(hasError) {
 								if (hasError)
 									$scope.tab = "review";
-								return hasError;
+								var document = $scope.document;
+								var nationalTargets = $scope.nationalTargets
+								var stopProcess = false;
+								var subRecordErrors = []
+								var subRecordsDialog = ngDialog.open({
+									template: 'subRecordsValidation.html',													
+									closeByDocument: false,
+									closeByEscape: false,
+									showClose: false,
+									closeByNavigation: false,
+									controller : ['$scope', function($scope){
+										var validationRequests = [];
+										if(document.targetPursued){
+											validationRequests = _.map(document.nationalTargets||[], "identifier");
+										}
+
+										if(document.progressAssessments && document.progressAssessments.length>0){
+											_.each(document.progressAssessments, function(a){
+												if(a.assessment){
+													validationRequests.push(a.assessment.identifier);
+												}
+											});
+										}
+										if(validationRequests.length > 0){
```

Comment:
Maybe this part could be extracted to a function reducing the size of this function?